### DISCUSSION TITLE
Overrides a list containing elements with quotes

### DISCUSSION THREAD

**Discussion by jayleicn** [facebookresearch/hydra#2820] [General]
How do we overide a list containing elements with quotes, the example below will raise `hydra.errors.OverrideParseException: no viable alternative at input '[date\='2023-10-10''`

$ python my_app.py debug=false "test.cfg_list=[date\='2023-10-10',set_name\=train]"
```
test:
  cfg_list:
    - date='2023-10-09'
    - set_name='val'
```

**Comment by odelalleau** [comment]
Use
```
"test.cfg_list=['date=\\'2023-10-10\\'','set_name=train']"
```

Explanation:
* Double quotes around everything so the shell doesn't interpret the []
* Single quotes around each element of your list because these strings contain the `=` character which is not allowed in unquoted strings in Hydra's overrides grammar
* Since you want a single quote in a single quoted string, you need to escape it in Hydra's overrides grammar, i.e. Hydra must see the string `\'`. The backslash character needs to be escaped for the shell ==> each single quote becomes `\\'`
